### ISSUE TITLE
Enhancement request: update model bindings after using conversion script

### ISSUE THREAD

**Issue by Arinomo** [OpenModelica/OpenModelica#8806]
### Description
Using the MWE in the comment [here](https://github.com/OpenModelica/OpenModelica/discussions/8726#discussioncomment-2410475) with a little modification, I tried to use MSL converting script using CLI. The conversion went ok, since i can simulate the converted model 'M', but the source code `TestConversion` itself does not change. Only after I manually update the model bindings, the source file was updated to new path.

With MSL 3.2.3 path:
```.mo
 package TestConversion
  model M
    Modelica.SIunits.Length L = 1;
  end M;

  function myFunction
  input Modelica.SIunits.Length x;
  output Modelica.SIunits.Length y;
  algorithm

  end myFunction;
  annotation(uses(Modelica(version="3.2.3")));
end TestConversion;
```

After conversion and update binding:
```.mo
package TestConversion
  model M
    Modelica.Units.SI.Length L = 1;
  end M;

  function myFunction
  input Modelica.Units.SI.Length x;
  output Modelica.Units.SI.Length y;
  algorithm

  end myFunction;
  annotation(uses(Modelica(version= "4.0.0")));
end TestConversion;
```

### Step to reproduce:
1. Open OMEdit with MSL 4.0.0 loaded
2. Open `TestConversion`package (That still used MSL 3.2.3 path)
3. Try to simulate model `M` and will fail (obviously)
4. Run `OpenModelica.Scripting.convertPackageToLibrary(TestConversion, Modelica, "4.0.0")`
5. Simulate model `M`to verify it was converted (should be successful)
6. Open `TestConversion`in text view -->  Still with MSL3.2.3 path
7. Update binding on Model `M`
8. Close and reopen `TestConversion`
9. source text is updated

### Expected behavior:
OMEdit automatically update binding after conversion to update the whole package.

### Version and OS
 - OpenModelica Version: v1.19.0-dev-590-g0897872ac1
 - OS: Windows 10, 64 bit

### Additional Context
The above example only true if `TestConversion` was saved as 'One File'-Package.

Package using Directory format can only be updated using save-as.

**Comment by casella** [comment]
@Arinomo, if you are using OMEdit the idea is not to use the CLI, but to use the GUI instead. Otherwise, the GUI may find itself in an inconsistent state, e.g. because of caching, but it's not its fault.

The workflow to do that is the following:
- open the TestLibrary using MSL 3.2.3
- right-click on it and select "Convert to newer version of used libraries"
- select version 4.0.0 of Modelica
- run the conversion

Once the conversion is complete, the library is saved (overwriting the previous version), the workspace is cleared, and then the library is reloaded with the new dependencies.

Can you please try that with the latest nightly and report if it works fine for you? Thanks!

**Comment by Arinomo** [comment]
@casella sorry for the very late answer. Sadly i still cant update a package that using my library via GUI.

The test packages are [here](https://github.com/OpenModelica/OpenModelica/files/8578065/ConversionTest.zip).

I manually copied ```myLib``` to ```~//AppData/Roaming/.openmodelica/libraries/``` and set OMEdit to automatically load it at start.

I the load ```myBib``` and right-click on it to update, but it seems OMEdit could not found ```myLib```

![Capture](https://user-images.githubusercontent.com/52284681/165638937-2f9eac62-b9de-4113-8766-65ab4257b749.JPG)

**Comment by casella** [comment]
I'll double-check. It could be that copying your library to the package manager directory is not enough, because it is not indexed, so that this feature only works for libraries that are actually managed by the package manager.

@sjoelund, @adeas31, can you please confirm?

**Comment by abuntrock** [comment]
@casella During testing of the BRSL at Bosch rexroth we also came across this issue. It does not seem to be possible to convert libraries if they are not part of the managed libraries (Cannot be installed via File->Managed Libraries->Install library). Our expectation would be that the OMEdit is able to convert any library, which is either loaded in the library tree or which is added to the package manager directory. From the status of this issue I assume that this is in Progress?

**Comment by sjoelund** [comment]
@perost this is a limitation of the approach to read the metadata from the package manager instead of parsing all libraries to get the conversion annotations, right?

We could have an API in OMEdit to create the metadata on user demand though.

**Comment by perost** [comment]
> @perost this is a limitation of the approach to read the metadata from the package manager instead of parsing all libraries to get the conversion annotations, right?
> 
> We could have an API in OMEdit to create the metadata on user demand though.

Yes, I think the conversion itself would work if called manually, but OMEdit relies on the package manager to find out which conversions are possible. It would be nice if user libraries could be added to the package manager, but we could probably also just load the library and read the conversion annotation if it's not in the package manager.

**Comment by casella** [comment]
> Yes, I think the conversion itself would work if called manually, but OMEdit relies on the package manager to find out which conversions are possible. It would be nice if user libraries could be added to the package manager, but we could probably also just load the library and read the conversion annotation if it's not in the package manager.

LGTM

**Comment by abuntrock** [comment]
Is there any update regarding this topic?

**Comment by abuntrock** [comment]
@casella  Is there a timeline when this will be possible?

**Comment by casella** [comment]
@abuntrock sorry, if this is useful to you we can try to prioritize it.

**Comment by abuntrock** [comment]
Since we will release our second fully OMEdit compatible BRSL version in the coming months, this issue is one of the most important ones (besides the performance tickets) from our perspective.

We see with other modelica tools, which are used for a longer time with the BRSL already, that many of our users have old models laying around that they want to open with a new BRSL version so that they have a starting point for a new project for example.

**Comment by abuntrock** [comment]
Is there any work planned regarding this ticket for OMEdit 1.26?

**Comment by casella** [comment]
I will prioritize this for 1.26.0

**Comment by abuntrock** [comment]
Since this did not make it into 1.26.0 and we now would have needed conversion for the first time with a Bosch external customer we wanted to ask if something is planned regarding this next year?
We would highly appreciate this.

**Comment by casella** [comment]
Will prioritize it for 1.27.0, sorry for not being able to include it in 1.26.0

**Comment by abuntrock** [comment]
@casella Thanks a lot. That would be a huge thing for us

**Comment by casella** [comment]
@perost, @adeas31 please prioritize this for January 2026. Thanks!

**Comment by perost** [comment]
It seems it's mostly a matter of figuring out which version of a non-managed library that's available to convert to. So if the package manager API doesn't return any results I guess OMEdit can just load the library, if it's not already loaded, check which version it is, and use that to call `convertPackageToLibrary`.

That only works for converting a model to the default version of the library though, if we want to be able to select a specific version to convert to I guess it becomes more complicated. But maybe that's not necessary?

**Comment by abuntrock** [comment]
@perost At least for us at Bosch Rexroth it would be sufficient if the action items described by @casella in https://github.com/OpenModelica/OpenModelica/issues/12681 are solved. 

I also think that converting to a library version which is not equal to the version of the library being loaded is not that important because I would assume that you always load the library in the version you want to convert your model to first.
